scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -538,6 +539,33 @@ def _check(self):
                 # This is also just for future-proofing.
                 assert isinstance(compat, str)
 
+                self._check_compatible(node, compat)
+        del self._checked_compatibles  # We have no need for this anymore.
+
+    def _check_compatible(self, node, compat):
+        if compat in self._checked_compatibles:
+            return
+
+        # The regular expression comes from dt-schema.
+        compat_re = r'^[a-zA-Z][a-zA-Z0-9,+\-._]+$'
```

Comment:
Should we stick to lowercase? That's what I was going on in my version of this, it matches what we have right now in the repository, would save some future incoherence with capitalization.

Reply:
I don't think so. The regular expression we are using here is the one that dt-schema itself is using to validate compatibles. Why should we be stricter?

Reply:
Because that is the style that we effectively adopted right now, the current set of compatibles (all 1200+ of them) are coherent in being all lowercase and the vendor names are all lowercase. It'd be nice to stick to it, avoid a code review conversation in the line of "but the standard allows it" versus "all the others are coherent".

Reply:
I checked upstream Linux and there are compatibles which use uppercase letters across multiple vendors (including big names like samsung, ti, fsl, brcm, intel, national, marvell). Given that, do you still think this is a good idea?

Reply:
Yeah I checked as well and honestly I find them to look odd, for example `"atmel,at25256B"`. I would find it nice to be strict and just stick to lowercase since we landed there naturally. Maybe seek for a third opinion? @galak?

Reply:
Another thought: if we try and be strict now and then it turns out we change our mind, it's easy to change the check to be less strict. Not so much the other way around.

Reply:
OK, I guess we can start there for now, but I'm going to resolve this comment and move the compatibles check to another PR now that we've got the tree-wide `s/_/-/g` change approved.